### ISSUE TITLE
[Request] Warn on void && Short circuit

### ISSUE THREAD

**Issue by pushkine** [microsoft/TypeScript#40151]
## Suggestion

Typescript should warn that `expr` will never be evaluated in the following 

```ts
void [condition] && [expr]
```

## Example

```ts
const bar = () => void condition && foo();
```
